scripts/db_error_check.j2

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Check prod service logs for errors indicating database update failures.
+
+# -- Internal variables -------------------------------------------------------
+
+script_name=$(basename "$0")
+
+prod_log="{{ tuxedo_logs_path }}/${USER}/prod.log"
+prod_log_lines="25"
+
+error_threshold="0"
+error_threshold_exceeded="false"
+
+# -- General ------------------------------------------------------------------
+
+check_user () {
+    if [[ "${USER}" == "root" ]]; then
+        echo "${script_name}: This script should be executed as a Tuxedo service user not root"
+        exit 1
+    fi
+}
+
+create_mail_file () {
+    mail_file=$(mktemp -t check_db_errors.XXXXXXX)
+    if [[ $? -ne 0 ]]; then
+        echo "Unable to create temporary mail file" >&2
+        exit 1
+    fi
+}
+
+check_db_errors () {
+    if [[ $(tail -${prod_log_lines} "${prod_log}" | grep -c "Return code:1200") -gt ${error_threshold} ]]; then
+        echo -e "prod.log file showing return code 1200 message; potential DB update issue, check log to investigate\n" >> "${mail_file}"
+        error_threshold_exceeded="true"
+    fi
+}
+
+# -- Alerts -------------------------------------------------------------------
+
+send_alert () {
+{% if alerts.enabled %}
+    mailx -r "${USER}@$(hostname)" -s "Return code 1200 message in prod.log; potential DB update issue - INVESTIGATE - $(date)" {% for recipient in alerts_config.db_error_check %}{{ recipient }} {% endfor %}< "${mail_file}"
+{% else %}
+    echo "Alerts disabled; no mail message(s) will be generated"
```

Comment:
It might be worth rewording this as it sounds more like an informational startup message rather than 'An attempt was made to send an alert' if I'm interpreting this correctly? Perhaps something like `An issue has been detected but alerting has been disabled` or similar?

Reply:
You're right, the message could read better here. I'll take a look at this separately as the existing text is consistent with several scripts in [ois-tuxedo-scripts](https://github.com/companieshouse/ois-tuxedo-scripts) so changes would be best made alongside similar changes there. The message is intended mainly for testing purposes, as alerts and stats will be enabled in live but disabled in other environments.